### PR TITLE
Make tlvmeta recognize feature map bitmaps.

### PR DESCRIPTION
**Pull request by andy31415**
zap does not support per cluster featuremap types because types are global, however we can use naming conventions to select the correct featuremap values.

This makes decoded data more user friendly.

Will replace FeatureMap decoding from an untyped bitmap to a typed bitmap with decoding.

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

File: src/lib/format/tests/TestDecoding.cpp [hunk 1]
```diff
@@ -274,7 +274,7 @@ void TestFullDataDecoding(nlTestSuite * inSuite, void * inContext)
                    "          endpoint_id: 0\n"
                    "          cluster_id: 49 == 'NetworkCommissioning'\n"
                    "          attribute_id: 65532 == 'featureMap'\n"
```
Comment (tcarmelveilleux) [inline src/lib/format/tests/TestDecoding.cpp:276]: `featureMap` is not the right name. `FeatureMap` is the attribute name

File: src/lib/format/tests/TestDecoding.cpp [hunk 2]
```diff
@@ -274,7 +274,7 @@ void TestFullDataDecoding(nlTestSuite * inSuite, void * inContext)
...
-                   "        NetworkCommissioning::featureMap: 4\n"
+                   "        NetworkCommissioning::featureMap: 4 == kEthernetNetworkInterface\n"
```
Comment (tcarmelveilleux) [inline src/lib/format/tests/TestDecoding.cpp:277]: The name of the feature in spec is EthernetNetworkInterface is `EthernetNetworkInterface` or `ET`. Just curious why `kXxxx` format is used.

**Review by tcarmelveilleux — APPROVED** [review]

**Review by jtung-apple — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]
Approved once @tcarmelveilleux 's comments are addressed

File: src/lib/format/tests/TestDecoding.cpp (hunk 1 above)
Comment (andy31415) [reply src/lib/format/tests/TestDecoding.cpp:276]: This is from the matter file. I believe we `lowerCamelCase` attribute names. We can update it if needed, but that seems a separate concern.

File: src/lib/format/tests/TestDecoding.cpp (hunk 2 above)
Comment (andy31415) [reply src/lib/format/tests/TestDecoding.cpp:277]: This is how we parse matter files. I think those use `kUpperCamelCase` convention. They can be adjusted/post processed, however this does seem a separate concern.

**Pull request merged**